src/services/atomic-swap.service.js

```diff
@@ -129,6 +129,43 @@ const removeAllWalletProposals = async walletId => {
   // TODO: Will also close the websocket channel
 };
 
+/**
+ @typedef {Object} AtomicSwapProposal
+ @property {string} proposalId - The unique identifier of the atomic swap proposal.
+ @property {string} partialTx - The partially constructed transaction of the atomic swap proposal.
+ @property {string | null} signatures - The signatures for the atomic swap proposal, if any.
+ @property {string} timestamp - The timestamp of when the atomic swap proposal last modified.
+ @property {number} version - The current version of the atomic swap proposal.
+ @property {Array} history - The history of the atomic swap proposal.
+ @property {string} history.partialTx - The partially constructed transaction at a specific point in
+ the proposal's history.
+ @property {string} history.timestamp - The timestamp of when the proposal reached the corresponding
+ partially constructed transaction in its history.
+ */
+
+/**
+ * Fetches the proposal data from the Atomic Swap Service, handling errors that may occur on the
+ * process
+ * @param {string} proposalId Proposal identifier on the Service
+ * @param {string} password Password, length 3 or more
+ * @returns {Promise<AtomicSwapProposal>} Returns the proposal identifier created by the service
+ */
+const serviceGet = async (proposalId, password) => {
+  if (!proposalId) {
+    throw new Error('Invalid proposalId');
+  }
+  if (!password || !password.length || password.length < 3) {
+    throw new Error('Password must have at least 3 characters');
+  }
```

Comment:
The service should reject creating a proposal with a weak password, right? Maybe it's not necessary to validate this here, the get will just return an error in case it's invalid. Can you identify the `get` error when it's an invalid ID, or invalid password?

Reply:
This was an attempt to reduce the amount of invalid calls the backend could receive, but not really necessary.
Refactored it on [ab27216](https://github.com/HathorNetwork/hathor-wallet-headless/pull/277/commits/ab2721629e4cc87ab47e71c4e0de5b86a480d1b3) to a simple "mandatory parameter check" in this case.

As for fully identifying valid ids and passwords, we could use _regexes_ to check for them, but I think it's too much complexity for almost zero benefit in this case: just avoiding one invalid `get` request to the service.